fix: keep sibling keywords when a referenced schema has a root-level $ref (#2965)
Redocly/redocly-cli: .changeset/root-ref-sibling-keywords.md

---
'@redocly/openapi-core': patch
'@redocly/cli': patch
---

> Fixed the `bundle` command losing schema keywords (such as `title`, `properties`, or `required`) written next to a `$ref` when the referenced schema starts with its own `$ref`.

Reviewer: ```suggestion
Fixed the `bundle` command losing schema keywords (such as `title`, `properties`, or `required`) written next to a `$ref` when the referenced schemas started with their own `$ref`.
```

Author: Applied in 232e3e304 — the changeset now reads "referenced schemas started with their own `$ref`".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

---
'@redocly/openapi-core': patch
'@redocly/cli': patch
---

Fixed the `bundle` command losing schema keywords (such as `title`, `properties`, or `required`) written next to a `$ref` when the referenced schema starts with its own `$ref`.

> These keywords are now kept in the bundled output, and the `lint` command validates them instead of skipping them.

Author: ```suggestion
```

Author: Applied in 113c97d5f — the trailing sentence was removed, so the changeset is now a single line.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

> Ref resolution now exposes such composed `$ref`s as a `chain` field on the resolve result, so custom rules and decorators can inspect them.

Author: This is not needed in changelog
```suggestion
```
